baker/GDriveImagesBaker.tsx

```diff
@@ -0,0 +1,134 @@
+import fs from "fs-extra"
+import path from "path"
+import { chunk } from "lodash"
+import * as db from "../db/db.js"
+import {
+    IMAGE_HOSTING_CDN_URL,
+    IMAGE_HOSTING_BUCKET_SUBFOLDER_PATH,
+} from "../settings/serverSettings.js"
+
+import {
+    ImageMetadata,
+    getFilenameAsPng,
+    retryPromise,
+} from "@ourworldindata/utils"
+import { Image } from "../db/model/Image.js"
+import sharp from "sharp"
+import { BAKED_BASE_URL } from "../settings/clientSettings.js"
+
+export const bakeDriveImages = async (bakedSiteDir: string) => {
+    // Get all GDocs images, download locally and resize them
+    const images: Image[] = await db
+        .queryMysql(
+            `SELECT * FROM images WHERE id IN (SELECT DISTINCT imageId FROM posts_gdocs_x_images)`
+        )
+        .then((results: ImageMetadata[]) =>
+            results.map((result) => Image.create<Image>(result))
+        )
+
+    const imagesDirectory = path.join(bakedSiteDir, "images", "published")
+
+    // If this causes timeout errors, try decreasing the chunk size (2 should be safe)
+    const imageChunks = chunk(images, 5)
```

Comment:
Actually, using a simple `chunk` is not the optimal outcome here since we're waiting for all 5 images to complete before we're starting to look at the next 5.

We can also change it to more of a pipeline situation:

```ts
import pMap from "p-map"

// ...

await pMap(images, mapFunction, {concurrency: 5})
```

Reply:
Yep, definitely better. Fixed.